src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -2145,6 +2148,68 @@ TEST_F(PolicyHandlerTest,
 }
 
 TEST_F(PolicyHandlerTest,
+       OnAppPermissionConsentInternal_ValidConnectionKey_FAIL) {
+  ChangePolicyManagerToMock();
+  const uint32_t device = 2u;
+
+  PermissionConsent permissions;
+  permissions.device_id = kDeviceId_;
+  permissions.consent_source = "consent_source";
+
+  policy::FunctionalGroupPermission group_permission_allowed;
+  CreateFunctionalGroupPermission(GroupConsent::kGroupAllowed,
+                                  kGroupAliasAllowed_,
+                                  kGroupNameAllowed_,
+                                  group_permission_allowed);
+
+  permissions.group_permissions.push_back(group_permission_allowed);
+
+  EXPECT_CALL(app_manager_, connection_handler())
+      .WillOnce(ReturnRef(conn_handler));
+  EXPECT_CALL(conn_handler, get_session_observer())
+      .WillOnce(ReturnRef(mock_session_observer));
+  EXPECT_CALL(mock_session_observer, GetDataOnDeviceID(device, _, NULL, _, _))
+      .WillOnce(Return(1u));
+
+  EXPECT_CALL(app_manager_, application(kConnectionKey_))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, policy_app_id()).WillOnce(Return(kPolicyAppId_));
+  EXPECT_CALL(*mock_app_, device()).WillOnce(Return(device));
+
+  sync_primitives::Lock wait_hmi_lock_first;
+  sync_primitives::AutoLock auto_lock_first(wait_hmi_lock_first);
+  WaitAsync waiter_first(kCallsCount_, kTimeout_);
+
+  EXPECT_CALL(*mock_policy_manager_, SetUserConsentForApp(_))
+      .WillOnce(NotifyAsync(&waiter_first));
+
+  ExternalConsentStatusItem item(1u, 1u, kStatusOn);
+  ExternalConsentStatus external_consent_status;
+  external_consent_status.insert(item);
+
+#ifdef EXTERNAL_PROPRIETARY_MODE
```

Comment:
@pold500 The changes was in  application_manager/src/policies/policy_handler.cc so I don't see any reasons to add ifdef here. Or you should add ifdef in policy_handler.cc.